### DISCUSSION TITLE
getting the permissions right

### DISCUSSION THREAD

**Discussion by ErnstBas** [GeriLife/companionship-care#209] [Q&A] [answered]
I am a bit stuck on #196.

Setting up the crud functionality with the django rest framework was relatively straightforward and it is working.

But being a newbie, I struggle adding the permissions. I am able to get the circle_id from the Activity.object, but despite various attempts no luck so far, checking this id with the Circle.object to verify whether the user in the request is a companion or organizer (or both). 

As an example this part:

```
    elif request.method == 'PUT':
        circle_id = Activity.objects.values_list("circle_id", flat=True).get(pk=id)
        circle = Circle.objects.get(id=circle_id)
        if request.user in circle.companions:
            serializer = ActivitySerializer(activity, data=request.data)
            if serializer.is_valid():
                serializer.save()
                return Response(serializer.data)        
            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 ```
This leads to an assertion error
>Expected a `Response`, `HttpResponse` or `HttpStreamingResponse` to be returned from the view, but received a `<class 'NoneType'>`
  
I am probably overlooking something , therefore some help would be very much appreciated.

**Comment by brylie** [comment, marked as answer]
From the code above, I don't see any default return for the `elif` statement. All returns are nested with `if` statements within the `elif`. This would make the default `elif` return to be `None`, as the error seems to indicate.

**Comment by ErnstBas** [comment]
Excellent, that resolved it.
I have PUT and DELETE working. I will continue with POST. 
Thanks again.

**Comment by ErnstBas** [comment]
Hello @brylie,
before doing a PR, I want to check some details:

Is it correct that GET requests are not needed? I added two of them anyway, just to make sure I could read the data. If not necessary, I will  delete this part.

To test the working of the API in the browser, I added session authentication in settings.py:
```
'rest_framework.authentication.SessionAuthentication',
```
and this path in urls.py:
```
path("api-auth/", include("rest_framework.urls", namespace = "rest_framework")
```
Please let me know if I need to delete this.

Thanks, Bas

**Reply by brylie** [reply]
Ah, good question. I updated the task checklist with an additional criteria.
